### PR TITLE
Update P2PUtils And Parameters From Altair

### PR DESCRIPTION
**Pull request by nisdas**
**What type of PR is this?**

Altair Work

**What does this PR do? Why is it needed?**

- [x] Updates our develop branch with altair specific parameters introduced.
- [x] Update branch with all the new methods introduced for p2putils.
- [x] Also fixes some bugs from config loading.

This updates the parameters to the latest ones introduced in the spec as of `v1.1.0-beta.2`. The biggest change is the change of
the target aggregators being increased to 16 from 4. This was introduced in `beta.1` .

For our changes in`p2putils`, the majority of the methods introduced are to help with fork versioning. These methods help determine the fork digest in a particular epoch, the fork digest in the next epoch, retrieving the fork data from a particular digest
and changes in our fork schedule structure to allow for a more accurate versioning scheme.

We also update our config with the merge and sharding specific parameters here to keep it in parity with how our current spec
configs are written. 

**Which issues(s) does this PR fix?**

N.A

**Other notes for review**

### CODE REVIEW

**Review by prestonvanloon — COMMENTED** [review]
Could you explain the changes in the PR description?

File: shared/params/mainnet_config.go [hunk 1]
```diff
@@ -208,7 +216,7 @@ var mainnetBeaconConfig = &BeaconChainConfig{
 	WeightDenominator:  64,
 
 	// Validator related values.
-	TargetAggregatorsPerSyncSubcommittee: 4,
+	TargetAggregatorsPerSyncSubcommittee: 16,
```
[the remark this answers is not in the document]
Comment (nisdas) [reply shared/params/mainnet_config.go:219]: This was changed here:
https://github.com/ethereum/eth2.0-specs/releases/tag/v1.1.0-beta.1

**Review by prestonvanloon — DISMISSED** [review]

File: shared/p2putils/fork.go [hunk 2]
```diff
@@ -2,15 +2,54 @@
...
+			break
+		}
+	}
+	return isForkEpoch, nil
+}
+
+// ForkDigestFromEpoch retrieves the fork digest from the current schedule determined
+// by the provided epoch.
+func ForkDigestFromEpoch(currentEpoch types.Epoch, genesisValidatorsRoot []byte) ([4]byte, error) {
+	if len(genesisValidatorsRoot) == 0 {
```
Comment (rauljordan) [inline shared/p2putils/fork.go:43]: Perhaps another zero root check

File: shared/p2putils/fork.go [hunk 3]
```diff
@@ -43,24 +82,83 @@ func CreateForkDigest(
...
+
+// NextForkData retrieves the next fork data according to the
+// provided current epoch.
+func NextForkData(currEpoch types.Epoch) ([4]byte, types.Epoch) {
+	fSchedule := params.BeaconConfig().ForkVersionSchedule
+	sortedForkVersions := SortedForkVersions(fSchedule)
+	nextForkEpoch := types.Epoch(math.MaxUint64)
+	nextForkVersion := [4]byte{}
+	for _, forkVersion := range sortedForkVersions {
+		epoch := fSchedule[forkVersion]
```
Comment (rauljordan) [inline shared/p2putils/fork.go:130]: I wonder if we should be safer here and check for existence in the map with `ok`. Same feedback for different accessors of this map

File: shared/p2putils/fork_test.go [hunk 4]
```diff
@@ -0,0 +1,365 @@
...
+	"time"
+
+	types "github.com/prysmaticlabs/eth2-types"
+	"github.com/prysmaticlabs/prysm/beacon-chain/core/helpers"
+	ethpb "github.com/prysmaticlabs/prysm/proto/prysm/v1alpha1"
+	"github.com/prysmaticlabs/prysm/shared/params"
+	"github.com/prysmaticlabs/prysm/shared/testutil/assert"
+)
+
+func TestFork(t *testing.T) {
```
Comment (rauljordan) [inline shared/p2putils/fork_test.go:16]: Add unknown fork test perhaps

**Review by rauljordan — APPROVED** [review]

**Pull request merged**